client/src/app/util/generateImage.js

```diff
@@ -16,35 +16,55 @@ const ENCODINGS = [
   'image/jpeg'
 ];
 
-const SCALE = 3;
+const INITIAL_SCALE = 3;
+const FINAL_SCALE = 1;
+const SCALE_STEP = 0.5;
+
+const DATA_URL_REGEX = /^data:((?:\w+\/(?:(?!;).)+)?)((?:;[\w\W]*?[^;])*),(.+)$/;
+
 
 export default function generateImage(type, svg) {
   const encoding = 'image/' + type;
 
-  let context,
-      canvas;
-
   if (ENCODINGS.indexOf(encoding) === -1) {
     throw new Error('<' + type + '> is an unknown type for converting svg to image');
   }
 
-  canvas = document.createElement('canvas');
+  const initialSVG = svg;
+
+  let dataURL = '';
+
+  for (let scale = INITIAL_SCALE; scale >= FINAL_SCALE; scale -= SCALE_STEP) {
+
+    let canvas = initializeCanvas();
 
+    svg = initialSVG.replace(/width="([^"]+)" height="([^"]+)"/, function(_, widthStr, heightStr) {
+      return `width="${parseInt(widthStr, 10) * scale}" height="${parseInt(heightStr, 10) * scale}"`;
+    });
 
-  svg = svg.replace(/width="([^"]+)" height="([^"]+)"/, function(_, widthStr, heightStr) {
-    return `width="${parseInt(widthStr, 10) * SCALE}" height="${parseInt(heightStr, 10) * SCALE}"`;
-  });
+    canvg(canvas, svg);
 
-  canvg(canvas, svg);
+    dataURL = canvas.toDataURL(encoding);
+
+    if (DATA_URL_REGEX.test(dataURL)) {
+      return dataURL;
+    }
+  }
+
+  // this will generate an invalid image if we end up here
+  return dataURL;

```

Comment:
Maybe, just an idea, we could raise an error here? It's something we would not expect and then, we could easily test it as well (expect error is thrown).

Reply:
Good idea, let me talk to Volker about this, if he's ok with that we could throw an error. That would make the test easier as well

Reply:
👍